### PR TITLE
PHSA URL changes to resolve AB#15023

### PR DESCRIPTION
**Pull request by ytqsl**
# Fixes or Implements [AB#15023](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15023)

[Task 15023](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/15023): PHSA API changes

## Description

Modified web alerts and file download urls to reflect PHSA changes

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by rayking29 — APPROVED** [review]

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs [hunk 1]
```diff
@@ -121,10 +121,9 @@ public async Task<RequestResult<EncodedMedia>> GetFileAsync(string hdid, string
                 RequestResult<PersonalAccount> response = await this.personalAccountsService.GetPatientAccountResultAsync(hdid).ConfigureAwait(true);
                 if (response.ResultStatus == ResultType.Success)
                 {
-                    string? pid = response.ResourcePayload?.PatientIdentity.Pid.ToString();
+                    Guid? pid = response.ResourcePayload?.PatientIdentity.Pid ?? throw new InvalidOperationException($"Pid not found for hdid {hdid}");
```
Comment (MikeLyttle) [inline Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:124]: I think it's possible to make the variable non-nullable now.

File: Apps/GatewayApi/src/Services/WebAlertService.cs [hunk 2]
```diff
@@ -76,10 +74,9 @@ public async Task DismissWebAlertsAsync(string hdid)
         {
             using Activity? activity = Source.StartActivity();
             this.logger.LogDebug("Sending request to dismiss web alerts to PHSA.");
-            PersonalAccount personalAccount = await this.personalAccountsService.GetPatientAccountAsync(hdid).ConfigureAwait(true);
-            string accountId = personalAccount.Id.ToString();
+            Guid? pid = await this.GetPersonalAccountPidByHdid(hdid).ConfigureAwait(true) ?? throw new InvalidOperationException($"No pid found for hdid {hdid}");
```
Comment (MikeLyttle) [inline Apps/GatewayApi/src/Services/WebAlertService.cs:77]: Same comment.

File: Apps/GatewayApi/src/Services/WebAlertService.cs [hunk 3]
```diff
@@ -88,11 +85,13 @@ public async Task DismissWebAlertAsync(string hdid, Guid webAlertId)
         {
             using Activity? activity = Source.StartActivity();
             this.logger.LogDebug("Sending request to dismiss web alert to PHSA.");
-            PersonalAccount personalAccount = await this.personalAccountsService.GetPatientAccountAsync(hdid).ConfigureAwait(true);
-            string accountId = personalAccount.Id.ToString();
+            Guid? pid = await this.GetPersonalAccountPidByHdid(hdid).ConfigureAwait(true) ?? throw new InvalidOperationException($"No pid found for hdid {hdid}");
```
Comment (MikeLyttle) [inline Apps/GatewayApi/src/Services/WebAlertService.cs:88]: Same comment.

**Review by sslaws — APPROVED** [review]

**Review by MikeLyttle — DISMISSED** [review]

**Review by rayking29 — DISMISSED** [review]

File: Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs (hunk 1 above)
Comment (ytqsl) [reply Apps/ClinicalDocument/src/Services/ClinicalDocumentService.cs:124]: Fixed, thanks. +10 for var.

**Review by MikeLyttle — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**